File: pyspark-demo/sales_filterer_fixed.py
#!/usr/bin/python3
from pyspark.sql import SparkSession
import os

spark = SparkSession \
    .builder \
    .appName("Sales Filterer Fixed") \
    .config("spark.jars.packages", "io.dataflint:spark_2.12:0.0.4-SNAPSHOT") \
    .config("spark.jars.repositories", "https://s01.oss.sonatype.org/content/repositories/snapshots") \
    .config("spark.plugins", "io.dataflint.spark.SparkDataflintPlugin") \
    .config("spark.ui.port", "11000") \
    .config("spark.eventLog.enabled", True) \
    .config("spark.sql.maxMetadataStringLength", "10000") \
    .master("local[1]") \
    .getOrCreate()

df = spark.read.load(os.getenv('SALES_FILES_LOCATION'))

df_filtered = df.filter(df.ss_quantity > 1).repartition("ss_quantity") # CHANGE

df_filtered.write \
    .mode("overwrite") \
    .partitionBy("ss_quantity") \
    .parquet("/tmp/store_sales")

input("sales filterer ended, press any key to continue..")
spark.stop()

File: spark-plugin/example_3_4_1/src/main/scala/io/dataflint/example/SalesFiltererFixed.scala
package io.dataflint.example

import org.apache.spark.sql.{SaveMode, SparkSession}

object SalesFiltererFixed extends App {
  val spark = SparkSession
    .builder()
    .appName("Sales Filterer Fixed")
    .config("spark.plugins", "io.dataflint.spark.SparkDataflintPlugin")
    .config("spark.ui.port", "10000")
    .config("spark.eventLog.enabled", true)
    .config("spark.sql.maxMetadataStringLength", "10000")
    .config("spark.dataflint.telemetry.enabled", false)
    .master("local[1]")
    .getOrCreate()

  import spark.implicits._

  spark.read
    .load("/Users/menishmueli/Documents/GitHub/spark-sql-perf/data/store_sales")
    .filter($"ss_quantity" > 1)
    .repartition($"ss_quantity")
    .write
    .mode(SaveMode.Overwrite)
    .partitionBy("ss_quantity")
    .parquet("/tmp/store_sales")

  // scala.io.StdIn.readLine()
  spark.stop()
}
